web/src/beta/lib/reearth-ui/components/DatePicker/index.tsx

```diff
@@ -0,0 +1,85 @@
+import { FC, useCallback, useEffect, useState, ChangeEvent } from "react";
+
+import { fonts, styled } from "@reearth/services/theme";
+
+export type DatePickerProps = {
+  value?: string;
+  disabled?: boolean;
+  size?: "normal" | "small";
+  onChange?: (text: string) => void;
+  onBlur?: (text: string) => void;
+};
+
+export const DatePicker: FC<DatePickerProps> = ({ value, disabled, onChange, onBlur }) => {
+  const [currentValue, setCurrentValue] = useState(value ?? "");
+  const [isFocused, setIsFocused] = useState(false);
+
+  useEffect(() => {
+    setCurrentValue(value ?? "");
+  }, [value]);
+
+  const handleChange = useCallback(
+    (e: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+      const newValue = e.currentTarget.value;
+      setCurrentValue(newValue ?? "");
+      onChange?.(currentValue);
+    },
+    [currentValue, onChange],
+  );
+
+  const handleBlur = useCallback(() => {
+    setIsFocused(false);
+    onBlur?.(currentValue);
+  }, [currentValue, onBlur]);
+
+  const handleFocus = useCallback(() => {
+    setIsFocused(true);
+  }, []);
```

Comment:
Do we need to wrap this function with useCallback?

Reply:
@ed0424 wrapping the function with useCallback helps in optimising performance and preventing unnecessary re-renders

Reply:
Consider the use case here both using or not using this useCallback is acceptable to me. We may have a big refactor after react v19 release. We can keep this for now i think.